components/public-api/gitpod/v1/configuration.proto

```diff
@@ -97,6 +98,14 @@ message UpdateConfigurationRequest {
   }
   message WorkspaceSettings {
     optional string workspace_class = 1;
+
+    // restricted_workspace_classes specifies the workspace classes that are NOT allowed to be used in this configuration.
+    // If empty, all workspace classes are allowed.
+    // Only update if update_restricted_workspace_classes is true.
+    repeated string restricted_workspace_classes = 2;
+
+    // update_restricted_workspace_classes specifies whether restricted_workspace_classes should be updated.
+    optional bool update_restricted_workspace_classes = 3;
```

Comment:
Agree with filip, it should be the other way around `allowed_workspaces_classes` if somehow another type is added it would should be disabled by default

Reply:
@filiptronicek @jeanp413  organization.proto doesn't have such problem is because we pass the old value always. It's not align to our guideline.

We need another field is because `repeated` can't be undefined or null for server side. I'll mark this as resolved

Reply:
>  it should be the other way around allowed_workspaces_classes if it's not null and somehow another type is added it should be disabled by default

this is still important to address

> repeated <type> <name> = 1;
> optional bool update_<name> = 2;

mmm feels error prone to forget about the linked property, how about creating a message: 

```
message MyRepeateadValue {
    repeated  string values 
}

message mainMessage {
    optional MyRepeateadValue optionalRepeated
}

```

Reply:
> > it should be the other way around allowed_workspaces_classes if it's not null and somehow another type is added it should be disabled by default
>
> this is still important to address

@jeanp413 I don't understand what this means.

`allowed_workspaces_classes` is not align to our PAPI guideline. So we don't need to compare with this while reviewing PR

Reply:
@jeanp413 Let's discuss proto changes in this thread https://github.com/gitpod-io/gitpod/pull/19480#discussion_r1505540915

resolving current thread